pyk/src/pyk/kcfg/semantics.py

```diff
@@ -34,6 +34,11 @@ def custom_step(self, c: CTerm) -> KCFGExtendResult | None: ...
 
     """Implement a custom semantic step."""
 
+    @abstractmethod
+    def is_mergeable(self, c1: CTerm, c2: CTerm) -> bool: ...
```

Comment:
This PR only introduces an interface. Its semantics will be determined by the merge_node function, which will be provided in another PR.